File: example/kotlin/src/main/java/pl/bratek20/othermodule/api/Properties.kt
package pl.bratek20.othermodule.api

import pl.bratek20.architecture.properties.api.PropertyKey

val OTHER_PROPERTY_KEY = PropertyKey("otherProperty")

data class OtherProperty(
    private val id: String,
    val name: String,
) {
    fun getId(): OtherId {
        return OtherId(this.id)
    }
}

File: example/typescript/main/OtherModule/Api/Properties.ts
namespace OtherModule {
    export const OTHER_PROPERTY_KEY = new PropertyKey("otherProperty")
}

class OtherProperty {
    private id = STRING
    name = STRING

    static create(
        id: string,
        name: string,
    ): OtherProperty {
        const instance = new OtherProperty()
        instance.id = id
        instance.name = name
        return instance
    }

    getId(): OtherId {
        return new OtherId(this.id)
    }
}